susemanager-utils/susemanager-sls/src/modules/uyuni_config.py

```diff
@@ -0,0 +1,1247 @@
+# coding: utf-8
+from typing import Any, Dict, List, Optional, Union, Tuple
+import ssl
+import xmlrpc.client  # type: ignore
+import logging
+
+import os
+import salt.config
+from salt.utils.minions import CkMinions
+import datetime
+
+log = logging.getLogger(__name__)
+
+__pillar__: Dict[str, Any] = {}
+__context__: Dict[str, Any] = {}
+__virtualname__: str = "uyuni"
+
+
+class UyuniUsersException(Exception):
+    """
+    Uyuni users Exception
+    """
+
+
+class UyuniChannelsException(Exception):
+    """
+    Uyuni channels Exception
+    """
+
+
+class RPCClient:
+    """
+    RPC Client
+    """
+
+    def __init__(self, user: str = None, password: str = None, url: str = "https://localhost/rpc/api"):
+        """
+        XML-RPC client interface.
+
+        :param user: username for the XML-RPC API endpoints
+        :param password: password credentials for the XML-RPC API endpoints
+        :param url: URL of the remote host
+        """
+
+        ctx: ssl.SSLContext = ssl.create_default_context()
+        ctx.check_hostname = False
+        ctx.verify_mode = ssl.CERT_NONE
+        self.conn = xmlrpc.client.ServerProxy(url, context=ctx, use_datetime=True, use_builtin_types=True)
+        if user is None or password is None:
+            # if user or password not set, fallback to default user defined on pillar data
+            if "xmlrpc" in (__pillar__ or {}).get("uyuni", {}):
+                rpc_conf = (__pillar__ or {})["uyuni"]["xmlrpc"] or {}
+                self._user: str = rpc_conf.get("user", "")
+                self._password: str = rpc_conf.get("password", "")
```

Comment:
Is it correct to say that this mechanism to provide username and password via pillar instead of parameters is a convenience option not to have to repeat them at many places in one's SLS file?

Reply:
That is partially correct. This is viewed as the default user and password to be used. However, if we setup multiple organizations we can only fallback to one user. I'm assuming this default user specified in the pillar data would be the server administrator.

Reply:
OK